testslide/strict_mock.py

```diff
@@ -476,6 +476,9 @@ def __init__(
         the attribute will change. Setting this value to False disables
         signature validation, and should only be used when type() is required
         to not change.
+        type_validation: use type hinting information from template to validate
+        that mock attribute types match them. Type validation also happens for
+        callable attributes (instance/static/class methods) calls.
```

Comment:
Maybe we can use some better formatted docstrings, google docstrings or reSt text eventually:
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
I'm all in to merging PRs for this.